[MaxEnt pipeline] Add feature class and regularization multipliers

frousseu opened the pull request:
This PR adds feature classes and regularization multipliers as user inputs. It also reorganizes the user inputs for a more logical order.

frousseu committed:
added feature classes and regularization multipliers as user inputs

frousseu committed:
reordered user inputs in a more logic way

frousseu committed:
added a check and an error for the feature classes chosen in accordance with ENMeval

frousseu commented:
@jamiemkass, perhaps you would be interested in reviewing this PR? If so, let me know and I will add you as a reviewer. If not, no worries!

jmlord commented:
Sorry, I see I created a conflict. The changes I made were cosmetic, so please ignore when merging and keep your changes.

jamiemkass reviewed (commented):
@frousseu Sorry, I just saw this notification. I think these are great changes, but a few comments. 

1. It would be good to show more than 2 example variables, as you are meant to run Maxent (and other ML SDMs) with many variables (perhaps 5 or so for the example?). 
2. I would recommend using "L", "LQ", and "LQH" as example fcs, because "P" is not recommended even by Phillips himself unless there is a strong reason to want interactions of all variables included. There is unfortunately no way to select individual interactions to put in, but users should be able to select "P" if they want. I'd also recommend removing "T" completely, as Phillips says it is mostly redundant with "H".
3. AIC does not use cross-validation results at all, so let's make sure that users understand that.

That's all from me. Thanks for making these changes.

-Jamie

jamiemkass commented:
Hello, trying to get back to this after a busy semester. Do we need another reviewer to pass this PR?

frousseu commented:
I completely forgot about that! I would look into it this week.

frousseu committed:
defaults to L, LQ LQH and added some predictors

frousseu commented:
@jamiemkass Hi! I added L, LQ and LQH as the default, but I have not removed the possibility for T (just to be consistent with what is generally available). I also added a couple of landcover predictors (the output SDM for the sugar maple is still not great though). For the AIC, I think the current pipeline uses AUC calculated from the cross-validation blocks and the metric used for tuning parameters is not given as an option in the current pipeline (though it is available in the scripts and could be made available). Do you think it is still worth it to mention your comment about the AIC or instead wait for when this metric is made available to the user? I'm also getting some error related to predictions  and dismo/predicts/terra stuff, but I don't have time to investigate right now. Some of the code may need to be updated in relation to dismo/predicts/terra...

JoryGriffith committed:
Merge branch 'main' into maxent_fcrm

jamiemkass commented:
@frousseu Sorry, I must have missed this update. Thanks for making it. I will try to investigate and see if I can fix the remaining issues.

jmlord committed:
fixed json error

JoryGriffith committed:
Fix

JoryGriffith committed:
Fixed pipeline

jmlord commented:
Hi,

If this branch already brings an improvement we could consider merging it, and document the remaining issues to fix in a future pull request.
About the terra errors: if it is due to a breaking change in the package, we could use conda to pin the package version. (like we did in Load from STAC script for example.)

jmlord committed:
Merge remote-tracking branch 'origin/main' into maxent_fcrm

jmlord committed:
cleanup after merge

jamiemkass commented:
@frousseu Thanks a lot. I will gladly review when the changes are done.

samaramanzin committed:
Merge branch 'main' into maxent_fcrm

samaramanzin committed:
updating json file

samaramanzin committed:
Updating metadata description

jmlord reviewed (commented):
Thanks for the description, much better! Here is what I found reading it:

Missing word: "SDM outputs be used" or maybe "SDM outputs are used"

Missing end of sentence: "calculating a
  range of biodiversity indicators including the Species Habitat Index (SHI),
  the Species Protection Index (SPI)"
maybe add "and Species Richness"?

This is for a single species, right? "by only including one occurrence per pixel" (if yes should mention "one occurrence of a single species per pixel")

Last paragraph: The first sentence could be removed and the rest sent inside the methods section along with the background point sentence.

"(see package vignette)" could use a markdown link here, I guess to send to this page? `(see [package vignette](https://jamiemkass.github.io/ENMeval/articles/ENMeval-2.0-vignette.html))`

Suggestion: "**In option,** Bias correction is achieved"

jmlord committed:
Example of parameter definition

samaramanzin committed:
Updating metadata description

jmlord commented:
@jamiemkass we will do a separate PR for the review once everything is ready. I think we still need to work on the documentation of inputs and outputs.

jamiemkass commented:
Great. Please let me know if you want one last review of the language.
